numpy/linalg/umath_linalg.c.src

```diff
@@ -3379,6 +3402,13 @@ static char lstsq_types[] = {
     NPY_CDOUBLE, NPY_CDOUBLE, NPY_DOUBLE, NPY_CDOUBLE, NPY_DOUBLE, NPY_INT, NPY_DOUBLE,
 };
 
+static char qr_types[] = {
+    NPY_FLOAT,   NPY_FLOAT,   NPY_FLOAT,
+    NPY_DOUBLE,  NPY_DOUBLE, NPY_DOUBLE,
+    NPY_CFLOAT,  NPY_FLOAT,  NPY_FLOAT,
+    NPY_CDOUBLE, NPY_DOUBLE, NPY_DOUBLE,
+};
```

Comment:
The first entry is for the type of input matrix and the second and third are for Q and R.